spec/validate.py

```diff
@@ -0,0 +1,199 @@
+"""
```

Comment:
This was `spec/test/validate.py`; I have moved it up a level (out of `test`) and refactored it.